pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -59,39 +59,27 @@ func (c *clusterRefreshController) onClusterChange(key string, cluster *mgmtv3.C
 		// not a KEv2 cluster
 		return cluster, nil
 	}
-
 	if !ready {
-		return cluster, nil
-	}
-
-	var lastRefreshTime string
-	var err error
-
-	if cluster.Annotations != nil {
-		lastRefreshTime = cluster.Annotations[clusterLastRefreshTime]
-	}
-
-	providerRefreshInterval, err := getProviderRefreshInterval(provider)
-	if err != nil {
-		return cluster, err
+		return c.updateCluster(cluster)
```

Comment:
This is to keep the flow consistent with the previous implementation. If the sync cron-job (the old one) ran before the cluster was ready, it would wait 5 minutes before checking it again. This achieves a similar result.

Reply:
Why update instead of re-enqueue ? (Also PR needs rebase)

Reply:
There is no need to re-enqueue because the the cluster will be automatically re-enqueued when the cluster is ready (that is, when the initial upstreamSpec is set).

As I see it, there are two options:
1. Keep it as is.
2. `return cluster, nil` here because there is nothing we can do until the upstreamSpec is set (and the cluster will be re-enqueued when it is set).

Reply:
Rebase is ready. I will push after I test and make any changes from your comments.

Reply:
Lets add a small comment here that explains we are updating refresh annotation since its not ready yet, otherwise lgtm

Reply:
I ended up choosing option (2) because, with some other changes that were introduced, it was causing way too many log messages.